textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/GramBooleanQuery.java

```diff
@@ -18,10 +18,10 @@
 		OR
 	}
 	QueryOp operator;
-	List<String> operandList;
-	List<GramBooleanQuery> subQueryList;
+	Set<String> operandSet;
+	Set<GramBooleanQuery> subQuerySet;
 	
-	int gramLength;
+	static int gramLength;
```

Comment:
RegexMatcher calls RegexToGramQueryTranslator.translate(), but RegexMatcher receives a RegexPredicate. The class that builds RegexPredicate determines gramLength, because gramLength should be consistent with the gramSize we use to build Lucene Analyzer.
So gramLength should be passed from the class that construct RegexPredicate ( Lucene Analyzer) and RegexMatcher. @zuozhi let's discuss this issue.